### DISCUSSION TITLE
Extract interface for ModuleClient, DeviceClient classes

### DISCUSSION THREAD

**Discussion by fgheysels** [Azure/azure-iot-sdk-csharp#2636] [General]
I think it would be great if there would exist an `IModuleClient` and `IDeviceClient` interface, which would then be implemented by the `ModuleClient` and `DeviceClient` classes.
This would allow developers to easily test code / classes that have a dependency on those classes.   Instead of directly injecting ModuleClient, one could inject `IModuleClient` and pass in a stub or mock in  unit-tests.

**Comment by drwill-ms** [comment]
We're working on a v2 of the SDKs and are enabling unit testable device clients.

**Reply by fgheysels** [reply]
That would be great!
Any idea of a release-date ?

**Reply by timtay-microsoft** [reply]
We have a [preview release](https://github.com/Azure/azure-iot-sdk-csharp/releases/tag/2022-11-03) out right now, but we don't have a release date set for the normal release just yet.
